src/core/common/logging.hpp

```diff
@@ -1500,6 +1500,31 @@ const char *otLogRegionToString(otLogRegion aRegion);
 #define _otPlatLog(aLogLevel, aRegion, aFormat, ...)                        \
     OPENTHREAD_CONFIG_PLAT_LOG_FUNCTION(aLogLevel, aRegion, aFormat, ## __VA_ARGS__)
 
+
+/*
+ * This provides a prototype for the platform defined logging function
+ *
+ * It is not intended to be a Doxygen comment because ...
+ * it is a macro that is used to override the normal function.
+ *
+ * The macro "OPENTHREAD_CONFIG_PLAT_LOG_FUNCTION" can be defined
+ * by three methods
+ *
+ * Method (A) via default 'openthread-core-default-config.h' as described above.
+ *
+ * Method (B) a platform specific override header that predefined the macro
+ *
+ * Method (C) on the compile command line, via a "-D" type define.
+ *
+ * In the "-D" case, there might not be a protytpe yet
+ * this just provides that missing prototype in the form required
+ *
```

Comment:
Internally, we where not sure how to describe the problem in clear way, making this very concise - does not explain enough, and making it too verbose makes the description too big.

The problem occurs in Case (C) - a function name is provided on the command line, but - in each source compilation unit, that function *does*not* yet have a prototype, but- when you look at this your first reading of this would look like we are creating a prototype for an ALL_CAPS function name, but in reality we are not.

For example, if somebody used:  -Dmy_log_function, what would provide the prototype?